cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -1261,9 +1297,10 @@ struct chunk_row_output_iter {
 struct start_offset_output_iterator {
   gpu::PageInfo* pages;
   int const* page_indices;
-  int cur_index;
-  int src_col_schema;
-  int nesting_depth;
+  size_t cur_index;
```

Comment:
Why `cur_index` type is changed? Do we really need `size_t`?

Reply:
`cur_index` is `size_t` since the index in `reduction_indices` is `size_t`